graylog2-web-interface/src/views/components/widgets/CopyToDashboardForm.jsx

```diff
@@ -0,0 +1,89 @@
+// @flow strict
+import React, { useEffect, useState, useCallback } from 'react';
+import { Modal, Button, ListGroup, ListGroupItem } from 'components/graylog';
+import type { DashboardsStoreState } from 'views/stores/DashboardsStore';
+import connect from 'stores/connect';
+import { PaginatedList, SearchForm } from 'components/common';
+import { DashboardsActions, DashboardsStore } from 'views/stores/DashboardsStore';
+
+type Props = {
+  onCancel: () => void,
+  onSubmit: (string, string) => void,
+  widgetId: string,
+  dashboards: DashboardsStoreState,
+};
+
+const CopyToDashboardForm = ({ widgetId, onCancel, dashboards: { list, pagination }, onSubmit }: Props) => {
+  const [selectedDashboard, setSelectedDashbaord] = useState('');
+  const [paginationState, setPaginationState] = useState({ query: '', page: 1, perPage: 5 });
+
+  const execSearch = useCallback((query, page, perPage) => {
+    DashboardsActions.search(query, page, perPage);
+  }, []);
+
+  const handleSearch = useCallback((query) => {
+    setPaginationState({
+      query,
+      ...paginationState,
+    });
+    setSelectedDashbaord('');
+    execSearch(query, paginationState.page, paginationState.perPage);
+  }, []);
+
+  const handleSearchReset = useCallback(() => handleSearch(''), []);
+
+  const handlePageChange = useCallback((page: number, perPage: number) => {
+    setPaginationState({
+      page,
+      perPage,
+      ...paginationState,
+    });
+    setSelectedDashbaord('');
+    execSearch(paginationState.query, page, perPage);
+  }, []);
+
+  useEffect(() => {
+    execSearch(paginationState.query, paginationState.page, paginationState.perPage);
+  }, []);
```

Comment:
I thought that a good idea, but it seems it does not work that way. I could not figure out completely why, but when I add `paginationState` here and remove `execSearch` from where `setPaginationState` is called the search will not executed. If I leave `execSearch` it will be called 2 times. One with the new state, and one with the old state. So no.